components/org.wso2.carbon.identity.application.authenticator.google/src/main/java/org/wso2/carbon/identity/application/authenticator/google/Utils.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.google;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.jwk.source.RemoteJWKSet;
+import com.nimbusds.jose.proc.BadJOSEException;
+import com.nimbusds.jose.proc.JWSKeySelector;
+import com.nimbusds.jose.proc.JWSVerificationKeySelector;
+import com.nimbusds.jose.proc.SecurityContext;
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTParser;
+import com.nimbusds.jwt.proc.ConfigurableJWTProcessor;
+import com.nimbusds.jwt.proc.DefaultJWTProcessor;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This is a utility classes for Google authenticator.
+ */
+public class Utils {
+
+    private static final List<String> ISSUER = Arrays.asList("https://accounts.google.com", "accounts.google.com");
+    private static final String JWS_RS256_URI = "https://www.googleapis.com/oauth2/v3/certs";
+    private static final String JWS_ES256_URI = "https://www.gstatic.com/iap/verify/public_key-jwk";
+
+    private Utils() {
+
+    }
+
+    /**
+     * This function validates the JWT token by its content using nimbus libraries.
+     *
+     * @param idToken  String. The jwt token string.
+     * @param audience String. Authenticator client ID to check validity.
+     * @return boolean. Validity of the JWT token returned via Google One Tap.
+     * @throws AuthenticationFailedException When JWT processor throws an exception.
+     */
+    public static boolean validateJWT(String idToken, String audience)
```

Comment:
Can we rename this method to highlight the specific functionality of the method?

Reply:
This is fixed 32c5c19af81429f5f8879cd01623cafa5cf37d45